pypiserver/plugin.py

```diff
@@ -0,0 +1,41 @@
+""" NOT YET IMPLEMENTED
+
+Plugins are callable setuptools entrypoints that are invoked at startup that
+a developer may use to extend the behaviour of pypiserver. A plugin for example
+may add an additional Backend to the system. A plugin will be called
+with the following keyword arguments
+
+* app: the Bottle App object
+* add_argument: A callable for registering command line arguments for your
+    plugin using the argparse cli library
+* backends: A Dict[str, callable] object that you may register a backend to.
+    The key is the identifier for the backend in the `--backend` command line
+    argument.
+    The callable must take a single argument `config` as a Configuration object
+    and return a Backend instance. It may be the class constructor or a factory
+    function to construct a Backend object
+
+In the future, the plugin callable may be called with additional keyword
+arguments, so a plugin should accept a **kwargs variadic keyword argument.
+"""
+from pypiserver.backend import SimpleFileBackend, CachingFileBackend
+from pypiserver import get_file_backend
+
+DEFAULT_PACKAGE_DIRECTORIES = ["~/packages"]
```

Comment:
Can we pull this from `config.DEFAULTS`?

Reply:
So in my opinion, if we're doing this as plugins, then the package root is a config setting specific to the file backends. Which would also mean that the cli argument should not be part of the global config, but provided by the plugin. Which is why I added that definition there, and not take it from the global config

Reply:
That does make sense. I just want to avoid needing to update this twice if we update it (although the likelihood that we'll ever change this is pretty low tbf)